### PR TITLE
First Commit

### PR DESCRIPTION
**Pull request by cs-shubham-morankar**
#### Description:

- New connector development, list of following actions has been implemented:
   - Get Locations List
   - Get Functions List
   - Get Function Details
   - Get Access Control Policy
   - Set Access Control Policy
- Not tested this connector, setup un-available (Paid services).

#### UTCs:

- [x] Basic Sanity test

### CODE REVIEW

File: google-cloud-functions/constants.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+"""
```
Comment (cs-omkar-kumbhar) [inline google-cloud-functions/constants.py]: this file can be removed if not used

File: google-cloud-functions/google_api_auth.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        if not config.get("redirect_url"):
+            self.redirect_url = DEFAULT_REDIRECT_URL
+        else:
+            self.redirect_url = config.get("redirect_url")
+
+    def convert_ts_epoch(self, ts):
+        datetime_object = datetime.strptime(ctime(ts), "%a %b %d %H:%M:%S %Y")
+        return datetime_object.timestamp()
+
+    def generate_token(self, REFRESH_TOKEN_FLAG):
```
Comment (cs-omkar-kumbhar) [inline google-cloud-functions/google_api_auth.py]: this function parameter name can be changed as there is constant declared in this file with same name

File: google-cloud-functions/google_api_auth.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                return "Bearer {0}".format(connector_config.get('accessToken'))
+            else:
+                logger.info("Token is valid till {0}".format(expires))
+                return "Bearer {0}".format(connector_config.get('accessToken'))
+        except Exception as err:
+            logger.error("{0}".format(str(err)))
+            raise ConnectorError("{0}".format(str(err)))
+
+
+def acquire_token(self, REFRESH_TOKEN_FLAG):
```
Comment (cs-omkar-kumbhar) [inline google-cloud-functions/google_api_auth.py]: this function parameter name can be changed as there is constant declared in this file with same name

File: google-cloud-functions/info.json [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+          "type": "text",
+          "visible": true,
+          "editable": true,
+          "required": true,
+          "tooltip": "Specify the name of the function based on which you want to retrieve access control policy from Google Cloud Functions."
+        },
+        {
+          "title": "Policy Version",
+          "name": "requestedPolicyVersion",
+          "description": "Specify the maximum policy version that will be used to format the policy. Valid values are 0, 1, and 3.",
```
Comment (cs-omkar-kumbhar) [inline google-cloud-functions/info.json:268]: these values can be specified in select type

File: google-cloud-functions/info.json [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+          "description": "Specify the comma-separated list of fully qualified names of fields.",
+          "type": "text",
+          "visible": true,
+          "editable": true,
+          "required": false,
+          "tooltip": "A FieldMask specifying which fields of the policy to modify. Only the fields in the mask will be modified. If no mask is provided, the following default mask is used: paths: \"bindings, etag\".",
+          "placeholder": "e.g. user.displayName,photo"
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-omkar-kumbhar) [inline google-cloud-functions/info.json:394]: output_schema is missing for all actions

File: google-cloud-functions/info.json (hunk 4 above)
Comment (cs-shubham-morankar) [reply google-cloud-functions/info.json:268]: We don't have exact words to put in select options. That's why we cannot change the field type.

File: google-cloud-functions/info.json (hunk 5 above)
Comment (cs-shubham-morankar) [reply google-cloud-functions/info.json:394]: We don't have setup access (paid services), this will release as uncertified.

**Review by cs-omkar-kumbhar — APPROVED** [review]

**Pull request merged**